pdr_backend/lake/lake_validate.py

```diff
@@ -0,0 +1,203 @@
+from typing import Dict, Tuple
+from enforce_typing import enforce_types
+
+import polars as pl
+
+from pdr_backend.lake.csv_data_store import CSVDataStore
+from pdr_backend.lake.table_bronze_pdr_predictions import (
+    bronze_pdr_predictions_table_name,
+)
+from pdr_backend.ppss.ppss import PPSS
+from pdr_backend.lake.lake_info import LakeInfo
+
+from pdr_backend.lake.gql_data_factory import GQLDataFactory
+from pdr_backend.lake.etl import ETL
+
+pl.Config.set_tbl_hide_dataframe_shape(True)
+
+
+@enforce_types
+class LakeValidate(LakeInfo):
+    def __init__(self, ppss: PPSS):
+        super().__init__(ppss)
+
+        self.config = {
+            "Validate table names in lake": self.validate_expected_table_names,
```

Comment:
Sure, I changed it to this.

Reply:
I reverted this because I believe it conflicts with your other feedback.

By returning a simple tuple(bool, str), I can count what's succeeded/failed, report accordingly, etc...

Reply:
it's not the same thing and there is no conflict with the other suggestion.
i.e. you have a config
```
self.config = {
   "table_names": {
         "violations": self.validate_expected_table_names.
         "default_success_message": "Tables in lake match..."
    }
}

in validate_expected_table names, instead of returning a Tuple:
if valid, return empty list
if not valid, return ["Tables in lake.... do not match..."]
```

then when displaying:
```
for value in config.values():
     if not value["violations"]:
           print value["default_success_message"]
           continue
           
     for violation in value["violations"]:
           print(violation)
```

This improves the following:
- `validate_` functions are only concerned with violation messages, don't manage success messages. They are also unaware of the result management itself. Just produce a list of errors.
- config does not contain spaces in key names!
- no need for the results variable (or rather we can rename "config" as results)
- move this part in generate() after initialising the lake, not in init. The "generate" function declares and fills the validation result entirely.

- also, on a personal note, tuples suck unless there's really a strong reason to describe data like that. The (bool, messages) structure that exists now has interdependency: the bool is actually a description of the nature of the messages, not a different characteristic of the data. Meaning there is reason to simplify it.

Reply:
i see, ok, i agree... "I was using a string" for a key, <rather than> a "key_string".
I could even use a list w/ functions to kiss.

============
my concern is that i wanted to avoid a "config", or too much indirection... or to have too much granularity with messages.

I just want a simple list of functions/tests/validations, that we run on the lake... and spend time making those good... not building the system around it.

i'm happy with just a list of functions.

i still think that a list[str, fn] where every fn returns tuple(bool, str), is a small, simple, good pattern.